### DISCUSSION TITLE
Message text is empty

### DISCUSSION THREAD

**Discussion by firasRihan** [telegraf/telegraf#1448] [Q&A]
hello I'm trying to extrat the text from this JSON : 
```
Context {
  update: {
    update_id: 367276573,
    message: {
      message_id: 1033,
      from: [Object],
      chat: [Object],
      date: 1620258983,
      text: '55'
    }
  }
}

```
using this code:

```
 bot.on('message', (message) =>{
         console.log(message);
          message.reply( message.text)
      })}
```
but all I'm getting is error: message text is empty.
Thanks!

**Comment by dcdunkan** [comment]
Oops. You got a problem there in your message handler! Actually, using the `bot.on("message")`'s function gives you a context object. And the message property is inside context object.

In your case, it should be like this:
```javascript
bot.on("message", (ctx) => {
    //            ^^ this ctx object contains a message property.
    // and the text property you are looking for is inside "that" message object.
    let text = ctx.message.text;
    ctx.reply(text);
});
```
In your code it is going to return `undefined`, because ctx object does not have a "text" property.
```
